network/api.go

```diff
@@ -96,11 +96,17 @@ type FreeNetworkRequest struct {
 // CreateNetworkRequest is sent by the daemon when a network needs to be created
 type CreateNetworkRequest struct {
 	NetworkID string
-	Options   map[string]interface{}
+	Options   CreateNetworkRequestOptions
 	IPv4Data  []*IPAMData
 	IPv6Data  []*IPAMData
 }
 
+type CreateNetworkRequestOptions struct {
+	Generic    map[string]string `json:"com.docker.network.generic"`
+	EnableIPv4 bool              `json:"com.docker.network.enable_ipv4"`
+	EnableIPv6 bool              `json:"com.docker.network.enable_ipv6"`
```

Comment:
I know we are used to have the structure field name match the JSON key, but I think that `ComDockerNetworkGeneric` would be a terrible name ^^

What do you think?

Reply:
If it is working as exxpected. LGTM